pyramid_oereb/contrib/print_proxy/mapfish_print.py

```diff
@@ -601,3 +601,34 @@ def _get_sorter(theme_code):
         sorter = Renderer._load_sorter(sorter_config['module'], sorter_config['class_name'])
         params = sorter_config.get('params', {})
         return sorter, params
+
+    def _get_sorted_legend(self, legend_list):
+        """
+        Takes an unsorted list of legend and returns it sorts, depending on the geometry type
+        and on the value of the geomety tye if this is available
+
+        Args:
+            legend_list: list of legends
+
+        Returns:
+             sorted list of legends
+        """
+        return sorted(legend_list, key=self._sort_legend_elem)
+
+    @staticmethod
+    def _sort_legend_elem(elem):
+        """
+        Returns a tuple of the geometry type of the element and the value of the geometry
+        type if it is available. Else it returns the geometry type and 0 as value
+
+        Args:
```

Comment:
Using a tuple as sorted-key is pretty advanced. I needed to look that up.
Since we know the data that gets sorted, I would go more into details here.

It sorts by geom_type, they are sorted alphabetically: AreaShare, LengthShare, NrOfPoints 
This turns out to be the desired order. If the data changes, alphabetically might not be enough anymore.

Second priority in ordering here is the area/length. I would also mention that in a comment. 
As explained in another comment, I think ascending order is better here.

Reply:
May be an even more explicit version:

```
#global variable 
LEGEND_ELEMENT_SORT_ORDER = { 'AreaShare': 1, 'LengthShare': 2, 'NrOfPoints': 3}

category = LEGEND_ELEMENT_SORT_ORDER.get(geom_type, 4) # everything else later
value = elem.get(geom_type, 0)

return (category, value)
```

Reply:
@pfirpfel do you think it makes sens to put the LEGEND_ELEMENT_SORT_ORDER in the project config? (https://github.com/openoereb/pyramid_oereb/blob/master/pyramid_oereb/standard/pyramid_oereb.yml.mako)

Reply:
project config is not necessary, there are currently no requirements for that.

Reply:
OK @jwkaltz thanks for the info.